src/app/server/DefaultFabricStorage.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/server/Server.h>
+#include <credentials/FabricTable.h>
+
+namespace chip {
+
+class DefaultFabricStorage : public FabricStorage
+{
+public:
+    DefaultFabricStorage(PersistentStorageDelegate & storage) : mStorage(storage) {}
+
+    CHIP_ERROR SyncStore(FabricIndex fabricIndex, const char * key, const void * buffer, uint16_t size) override
+    {
+        return mStorage.SyncSetKeyValue(key, buffer, size);
+    };
+
+    CHIP_ERROR SyncLoad(FabricIndex fabricIndex, const char * key, void * buffer, uint16_t & size) override
+    {
+        return mStorage.SyncGetKeyValue(key, buffer, size);
+    };
+
+    CHIP_ERROR SyncDelete(FabricIndex fabricIndex, const char * key) override { return mStorage.SyncDeleteKeyValue(key); };
+
+private:
+    PersistentStorageDelegate & mStorage;
+};
+
+FabricStorage & __attribute__((weak)) Server::GetFabricStorage()
```

Comment:
Instead of using a weak symbol, can we just have an API for applications to call that sets the delegate to use?

Reply:
I believe longer term we would want all apps to inject, so I believe a delegate + some default (compile time flag to enable default) may work.

Reply:
The correct way, overall, would be to inject it in Server::Init(), from application, or alternatively get rid of Server class altogether and find a **well-discussed, consensus** solution to the problem rather than more incremental stuff that is hard to document/discover

Reply:
Furthermore, we should replace FabricStorage with just PersistentStorageDelegate usage within the FabricTable, since the code in there already does this